sale_coupon_most_expensive/models/sale_order.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+from odoo.tools import float_compare
+
+
+class SaleOrder(models.Model):
+    """Extend to implement most_expensive_product disc option."""
+
+    _inherit = "sale.order"
+
+    def _get_most_expensive_line(self):
+        # Mimic _get_cheapest_line method.
+        lines = self.order_line.filtered(
+            lambda r: not r.is_reward_line
+            and float_compare(r.price_reduce, 0, precision_digits=2) > 0
```

Comment:
You cannot hardcode precision_digits like that. You need to get value from 'Product Price' as defined in field : https://github.com/odoo/odoo/blob/13.0/addons/sale/models/sale.py#L1258

Reply:
@rousseldenis Thanks for your feedback: it's now fixed.